### DISCUSSION TITLE
Fastify example

### DISCUSSION THREAD

**Discussion by nesro** [node-oauth/node-oauth2-server#300] [Q&A]
Hi all,

I want to create an OAuth server in NestJS + Fastify. This library seems to be the most updated and is framework-agnostic.

Is there any example that could help me? I did try to find something, but no luck.

If I will have to write my own "wrapper" for fastify (because there isn't any), do you have any tips or pitfalls to be aware of?

Thanks, Tomas

**Comment by jankapunkt** [comment]
Hi @nesro thanks for interest in this library. I personally have no experience with fastify. However, if you have some background in express then you might get the idea for a reference implementation by reading the code of the [express wrapper](https://github.com/node-oauth/express-oauth-server).

This could be a first entrypoint for inspiration. You can also check out the tests in this repo (there are unit tests, integration tests and compliance tests) to get an idea of what to be aware of when dealing with certain use-cases.

Finally I hope our documentation is detailed enough to provide you with enough context. If there are any things unclear, confusing or ambiguous, please let us know so we can improve on it.
